addons/project/models/project.py

```diff
@@ -0,0 +1,832 @@
+                "   them or by someone of their company\n"
+                "- Employees Only: employees see all tasks or issues\n"
+                "- Followers Only: employees see only the followed tasks or issues; if portal\n"
+                "   is activated, portal users see the followed tasks or issues.")
+    state = fields.Selection([('draft','New'),
+                              ('open','In Progress'),
+                              ('cancelled', 'Cancelled'),
+                              ('pending','Pending'),
+                              ('close','Closed')],
+                              'Status', required=True, copy=False, default='open')
+    doc_count = fields.Integer(compute='_compute_get_attached_docs', string="Number of documents attached")
+    date_start = fields.Date('Start Date')
+    date = fields.Date('Expiration Date', index=True, track_visibility='onchange')
+    use_tasks = fields.Boolean(related='analytic_account_id.use_tasks', default=True)
+
+    def _compute_get_attached_docs(self):
+        Attachment = self.env['ir.attachment']
+        for project in self:
+            project_attachments = Attachment.search_count([('res_model', '=', 'project.project'), ('res_id', '=', project.id)])
+            task_attachments = Attachment.search_count([('res_model', '=', 'project.task'), ('res_id', 'in', project.tasks.ids)])
+            project.doc_count = project_attachments + task_attachments
+
+    def _compute_task_count(self):
+        for project in self:
+            project.task_count = len(project.task_ids)
+
+    def _compute_task_needaction_count(self):
+        projects = self.env['project.task'].read_group([('project_id', 'in', self.ids), ('message_needaction', '=', True)], ['project_id'], ['project_id'])
+        mapped_data = {project['project_id'][0]: int(project['project_id_count']) for project in projects}
+        for project in self:
+            project.task_needaction_count = mapped_data.get(project.id)
+
+    @api.constrains('date_start', 'date')
+    def _check_dates(self):
+        for project in self:
+            if project.date_start and project.date and project.date_start > project.date:
+                raise UserError(_('Error! project start-date must be lower than project end-date.'))
+
+    @api.onchange('partner_id')
+    def _onchange_partner_id(self):
+        if 'pricelist_id' in self.fields_get():
+            self.pricelist_id = self.partner_id.property_product_pricelist
+
+    @api.model
+    def create(self, vals):
+        ir_values = self.env['ir.values'].get_default('project.config.settings', 'generate_project_alias')
+        if ir_values:
+            vals['alias_name'] = vals.get('alias_name') or vals.get('name')
+        create_context = dict(self.env.context,
+                              project_creation_in_progress=True,
+                              mail_create_nosubscribe=True)
+        return super(Project, self.with_context(create_context)).create(vals)
```

Comment:
Why not just use `self.with_context(project_creation_in_progress=True, mail_create_nosubscribe=True)` instead of manually copying the existing context?